src/array/list/mutable.rs

```diff
@@ -142,6 +142,22 @@ impl<O: Offset, M: MutableArray> MutableListArray<O, M> {
         Self::new_from(values, data_type, capacity)
     }
 
+    /// Creates a new [`MutableListArray`] from a [`MutableArray`], [`Offsets`] and
+    /// [`MutableBitmap`].
+    pub fn new_from_mutable(
+        values: M,
+        offsets: Offsets<O>,
+        validity: Option<MutableBitmap>,
+    ) -> Self {
+        let data_type = values.data_type().clone();
```

Comment:
LGTM, but better  check that the `values.len()` equals the last element of offsets.